security-framework/src/key.rs

```diff
@@ -119,6 +121,19 @@ impl SecKey {
         }
     }
 
+    /// Returns the programmatic identifier for the key. For keys of class 
+    /// kSecAttrKeyClassPublic and kSecAttrKeyClassPrivate, the value is the 
+    /// hash of the public key.
+    #[cfg(any(feature = "OSX_10_12", target_os = "ios"))]
+    pub fn application_label(&self) -> Vec<u8> {
+        let attributes = self.attributes();
+        unsafe {
+            CFData::wrap_under_get_rule(
+                attributes.get(kSecAttrApplicationLabel.to_void()).cast()
```

Comment:
Is the attribute guaranteed to exist? Could get return nil?

Reply:
```swift
var params: [String: Any] = [
    kSecAttrKeyType as String: kSecAttrKeyTypeAES,
    kSecAttrKeySizeInBits as String: 256,
    kSecAttrIsPermanent as String: false,
]

var error: Unmanaged<CFError>?
guard let key = SecKeyGenerateSymmetric(params as CFDictionary, &error) else {
    throw error!.takeRetainedValue() as Error
}
let attrs = SecKeyCopyAttributes(key)! as NSDictionary
attrs[kSecAttrApplicationLabel]!
```

The attribute is guaranteed to exist for "permanent" keys, however it does not appear to exist non-permanent keys as demonstrated by the above code snippet.

I've changed the return type to be `Option<Vec<u8>>` and used `.find` instead of `.get` on the attributes dictonary.